pycqed/analysis_v2/timedomain_analysis.py

```diff
@@ -290,17 +400,271 @@ def extract_data(self):
+            for qbn in self.qb_names:
+                if not len(self.cal_states_dict[qbn]) or \
+                        'pca' in self.rotation_type[qbn].lower():
+                    self.data_to_fit[qbn] = self.rotation_type[qbn]
+                else:
+                    csr = [(k, v) for k, v in
+                           self.cal_states_rotations[qbn].items()]
+                    csr.sort(key=lambda t: t[1])
+                    self.data_to_fit[qbn] = f'p{csr[-1][0]}'
+
+        # TODO: Steph 15.09.2020
+        # This is a hack to allow list inside data_to_fit.
+        for qbn in self.data_to_fit:
+            if isinstance(self.data_to_fit[qbn], (list, tuple)):
+                self.data_to_fit[qbn] = self.data_to_fit[qbn][0]
+
+    def get_data_filter(self):
+        """
+        Extracts the data_filter parameter from the options_dict or metadata
+        and assigns it a reasonable default value based on the information in
+        self.prep_params.
+
+        Creates the following attribute:
+            - self.data_filter: function that will be used in
+                create_meas_results_per_qb to filter/process the data
+            - self.data_with_reset: bool that is checked in
+                prepare_raw_data_plots
+        """
+        # flag to be used in prepare_raw_data_plots
+        self.data_with_reset = False
+        self.data_filter = self.get_param_value('data_filter')
+        if self.data_filter is None:
+            if 'active' in self.prep_params.get('preparation_type', 'wait'):
+                reset_reps = self.prep_params.get('reset_reps', 1)
+                self.data_filter = lambda x: x[reset_reps::reset_reps+1]
+                self.data_with_reset = True
+            elif "preselection" in self.prep_params.get('preparation_type',
+                                                        'wait'):
+                self.data_filter = lambda x: x[1::2]  # filter preselection RO
+            else:
+                self.data_filter = lambda x: x
+
     def create_sweep_points_dict(self):
+        """
+        Creates self.proc_data_dict['sweep_points_dict'][qbn]['sweep_points']
+        containing the hard sweep points (1st sweep dimension).
+        It can be created from the following given parameters (the priority in
+        which these parameters are considered is the following):
+            - 1D sweep points taken from a SweepPoints instance or its repr
+            based on the meas_obj_sweep_points_map or main_sp (used with split
+            data, see self.split_data).
+            ! If SweepPoints are given then the meas_obj_sweep_points_map must
+            also be specified !
+            - sweep_points_dict of the form {qbn: swpts_1d_array}
+            - swpts_1d_array from hard_sweep_params of the form
+            {sweep_param_name: {'values': swpts_1d_array, 'unit': unit_str}}
+            - self.raw_data_dict['hard_sweep_points'] created by the base class
```

Comment:
General comment: there are cases where 1st dim sweep points are not hard sweep points (mixer calib, which will now also be analyzed with v2). At some point, we might want to clean up the naming conventions. (Not part of this PR.)

Reply:
👍